### ISSUE TITLE
Static Content gallery: Theme based on Gallery Style field value

### ISSUE THREAD

**Issue by aaronschachter** [DoSomethingArchive/legacy-website#2749]
Refs #2646 - blocked by #2748

`dosomething_static_content_preprocess_gallery_vars` is now adding a `style` variable for each Gallery.

`default`: Should be rendered as is.

`2col`: The 2 column style as per the Closed Campaign

`3col_short`: The 3 column style per the Press gallery

These values are defined on the `field_gallery_style` content type.

**Comment by barryclark** [comment]
For my own reference: Need to be able to use the 3 different gallery types used here https://www.dosomething.org/campaigns/comeback-clothes

**Comment by barryclark** [comment]
Shooting you over this one, @tjrosario. @aaronschachter will connect with you on the details of it. cc @mikefantini

**Comment by mikefantini** [comment]
@tjrosario when trying to add the 3 column short gallery to static content, the gallery isn't pulling in images that are uploaded as the thumbnail crop.

see http://staging.beta.dosomething.org/about/fantiniz-partnerz as an example

using this issue since it's still open to report the bug rather than creating a new issue

let me know if you need any other info

**Comment by tjrosario** [comment]
@aaronschachter @angaither the `image_url` field is returning blank for the gallery photos on static pages.  Would one of you be able to take a look or point me in the direction for a fix?

**Comment by angaither** [comment]
@tjrosario which field are you trying to use? what the nid?

**Comment by tjrosario** [comment]
![image](https://cloud.githubusercontent.com/assets/1566749/3679516/185873f6-12a5-11e4-9e05-371a4319e235.png)

NID is 1292

**Comment by angaither** [comment]
That image only has something in the `field_thumbnail` is that what you're trying to grab?

**Comment by aaronschachter** [comment]
For the "3column short" gallery - it is supposed to display the Thumbnail field. So, we need to alter the prrocess, right now it's hardcoded to use Square:

https://github.com/DoSomething/dosomething/blob/dev/lib/modules/dosomething/dosomething_static_content/dosomething_static_content.module#L109

If the gallery style is default, keep this code as is, passing `'square'` as the `'ratio'` parameter to `dosomething_image_get_themed_image`.

```
    $vars['galleries'][$i]['items'][$a]['image'] = dosomething_image_get_themed_image($gallery_item['field_gallery_image']['#items'][0]['target_id'], 'square', '300x300');
```

If NOT default, 'thumb'  should be passed:

```
    $vars['galleries'][$i]['items'][$a]['image'] = dosomething_image_get_themed_image($gallery_item['field_gallery_image']['#items'][0]['target_id'], 'thumb', 'whatever size goes here');
```

@tjrosario I'm working on getting my local environment set up again, but i can possibly jump in here. Ideally setting the style to the Neue specific classes should happen within the theme and not module too..

**Comment by mikefantini** [comment]
@aaronschachter so is this issue on you now?

**Comment by aaronschachter** [comment]
Opened up #2875 to unblock @tjrosario on grabbing the Thumbnail field for the non-default galleries.  Per https://github.com/DoSomething/dosomething/pull/2875#issuecomment-49960715 there seemed to be css issues preventing us from closing out the issue, need @tjrosario to give a look.

**Comment by aaronschachter** [comment]
@tjrosario I've added Images and different Galleries / Gallery Styles to http://staging.beta.dosomething.org/about/fantiniz-partnerz to give examples of the CSS bug.  Seems there's also a bug in my code that's rendering the 3rd standard gallery i've added in the 2 column style: investigating.

**Comment by aaronschachter** [comment]
The bug in my code has been fixed, but @tjrosario there are still CSS issues with the `-duo` gallery: http://staging.beta.dosomething.org/about/fantiniz-partnerz

**Comment by mikefantini** [comment]
@tjrosario @weerd re-opening.

Gallery Style is working properly on campaign runs (when done in conjunction with Gallery Type?) but is not working on Static Content pages on it's on.

Working: http://staging.beta.dosomething.org/campaigns/comeback-clothes
Not Working: http://staging.beta.dosomething.org/about/fantiniz-partnerz
